bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,248 @@
+                        break;
+                    case "cs":
+                        response.setLastCommandStatus(EtherRainCommandStatus.valueOf(status.toUpperCase()));
+                        break;
+                    case "rz":
+                        response.setLastCommandResult(EtherRainCommandResult.valueOf(status.toUpperCase()));
+                        break;
+                    case "ri":
+                        response.setLastActiveValue(Integer.parseInt(status));
+                        break;
+                    case "rn":
+                        response.setRainSensor(Integer.parseInt(status) == 1);
+                        break;
+                    default:
+                        logger.debug("Unknown response: {}", command);
+                }
+            }
+        }
+
+        return response;
+    }
+
+    public synchronized boolean commandIrrigate(int delay, int zone1, int zone2, int zone3, int zone4, int zone5,
+            int zone6, int zone7, int zone8) {
+        try {
+            sendGet("result.cgi?xi=" + delay + ":" + zone1 + ":" + zone2 + ":" + zone3 + ":" + zone4 + ":" + zone5 + ":"
+                    + zone6 + ":" + zone7 + ":" + zone8);
+        } catch (IOException e) {
+            logger.warn("Could not send irrigate command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public synchronized boolean commandClear() {
+        try {
+            sendGet("/result.cgi?xr");
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public synchronized boolean commandLogin() throws EtherRainException {
+        try {
+            sendGet("/ergetcfg.cgi?lu=" + ETHERRAIN_USERNAME + "&lp=" + password);
+        } catch (IOException e) {
+            logger.warn("Could not send clear command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public synchronized boolean commandLogout() {
+        try {
+            sendGet("/ergetcfg.cgi?m=o");
+        } catch (IOException e) {
+            logger.warn("Could not send logout command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    private synchronized List<String> sendGet(String command) throws IOException {
+        String url = "http://" + address + ":" + port + "/" + command;
+
+        ContentResponse response;
+
+        try {
+            response = httpClient.newRequest(url).timeout(HTTP_TIMEOUT, TimeUnit.SECONDS).send();
+            if (response.getStatus() != HttpURLConnection.HTTP_OK) {
+                logger.warn("Etherrain return status other than HTTP_OK : {}", response.getStatus());
+                return new LinkedList<String>();
```

Comment:
If you don't expect this return value to be modified by the callers then return a Collections.emptyList() instead.